src/tqdm_publisher/_progress_subscriber.py

```diff
@@ -2,13 +2,6 @@
 
 
 class TQDMProgressSubscriber(TQDMPublisher):
-    def __init__(self, iterable, *, announcer: "?", request_id: str, message: dict, **tqdm_kwargs):
+    def __init__(self, iterable, on_progress_update: callable, **tqdm_kwargs):
         super().__init__(iterable, **tqdm_kwargs)
-
-        def on_progress_update(format_dict) -> None:
-            """
-            Describe what this announcer is all about...
-            """
-            announcer.announce(dict(request_id=request_id, **message))
-
-        self.subscribe(callback=on_progress_update)
+        self.subscribe(lambda format_dict: on_progress_update(dict(progress_bar_id=self.id, format_dict=format_dict)))
```

Comment:
Only because there was never anything called an `announcer` here, and callbacks are a generic way to notify another part of code about the changes here. Whereas passing an announcer as an argument assumes a certain workflow for the user, as implemented here, and ultimately runs / provides a value that will run a callback anyways